File: 0118 pascals-triangle/solution.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        IList<IList<int>> ans = new List<IList<int>>();
        for(int i=0;i<numRows;i++) {
            IList<int> l = new List<int>();
            for(int j=0;j<=i;j++) {
                if(j==0 || j==i) {
                    l.Add(1);
                } else {
                    int n=ans[i-1][j]+ans[i-1][j-1];
                    l.Add(n);
                }
            }
             ans.Add(l);
        }
        return ans;
    }
}

File: 0118 pascals-triangle/solution.rs
impl Solution {
    pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {
        let mut ans : Vec<Vec<i32>> = Vec::new();
        for i in 0..num_rows {
            let mut li: Vec<i32> = Vec::new();
            for j in 0..=i {
                if j==0 || j==i {
                    li.push(1);
                } else {
                    let ans = ans[(i-1) as usize][j as usize] + ans[(i-1) as usize][(j-1) as usize];
                    li.push(ans);
                }
            }
            ans.push(li);
        }
        ans
    }
}
